EIPS/eip-6404.md

```diff
@@ -0,0 +1,1000 @@
+---
+eip: 6404
+title: Transition to SSZ
+description: Unifies the CL and EL block representations
+author: Etan Kissling (@etan-status)
+discussions-to: https://ethereum-magicians.org/t/eip-6404-transition-to-ssz/12783
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-01-30
+requires: 155, 658, 1559, 2718, 2930, 4844, 4895
+---
+
+## Abstract
+
+This EIP aligns the definition of the EL block with the CL `ExecutionPayload` structure.
+
+## Motivation
+
+While the CL `ExecutionPayload` and the EL block structure map to each other, they are differently encoded. This EIP aims to align the encoding between them, taking advantage of the more modern SSZ format used by the CL. This brings several advantages:
+
+1. **Reducing complexity:** Merkle-Patricia Tries (MPT) are hard to work with. Replacing them with SSZ leaves only the state trie in the legacy MPT format.
+
+2. **Better for smart contracts:** The SSZ format is optimized for production and verification of merkle proofs. It allows proving specific fields of containers and allows chunked processing, e.g., to support handling transactions that do not fit into calldata.
+
+3. **Better for light clients:** Light clients with access to the CL `ExecutionPayload` no longer need to obtain the matching EL block header to verify proofs rooted in `transactions_root` or `withdrawals_root`.
```

Comment:
or receipts_root?

Reply:
The `receipts_root` is represented as an MPT across both CL and EL. This discrepancy only exists for `transactions_root` and `withdrawals_root`.